src/components/uri/Visualization.tsx

```diff
@@ -437,11 +426,15 @@ export class VisualizationWrapped extends React.Component<
         const visualizationClassUri: string = MdObjectHelper.getVisualizationClassUri(mdObject);
         const visualizationClass = await fetchVisualizationClass(this.sdk, visualizationClassUri);
         const featureFlags = await getFeatureFlags(this.sdk, projectId);
+        const localeFromUserAccount = await fetchLocalizationFromUser(this.sdk, projectId);
```

Comment:
This is much better, nice job! :) One improvement that could be made with not much of an effort would be to use `Promise.all` to perform some of these `await` operations in parallel (you can definitely parallelize `fetchVisualizationClass`, `getFeatureFlags` and `fetchLocalizationFromUser`, but maybe even others). So if you have the time, please try looking into this, it could make the component quite significantly faster.

Reply:
Makes sense. I completely forgot about Promise.all 👍